baricadr/model/backends.py

```diff
@@ -184,6 +190,68 @@ def restricted_walk(self, path, max_depth):
     def temp_rclone_config(self):
         raise NotImplementedError()
 
+    def do_remote_tree(self, repo, path, max_depth=1, backend_specific_options=""):
+        """
+        List content in a distant path, with missing files tagged with a '*'
+        """
+        tempRcloneConfig = self.temp_rclone_config()
+
+        rel_path = repo.relative_path(path)
+
+        src = "%s:%s%s" % (self.name, self.remote_prefix, rel_path)
+
+        max_depth_command = ""
+
+        try:
+            max_depth = int(max_depth)
+        except ValueError:
+            max_depth = 1
+        # If not 0 (0 is for listing all)
+        if max_depth:
+            max_depth_command = "--max-depth " + str(max_depth)
+
+        cmd = "rclone lsjson -R --config '%s' '%s' %s %s" % (tempRcloneConfig.name, src, backend_specific_options, max_depth_command)
+        current_app.logger.info(cmd)
+        p = Popen(cmd, shell=True, stdin=PIPE, stdout=PIPE, stderr=PIPE)
+        output, err = p.communicate()
+        retcode = p.returncode
+        try:
+            json_output = json.loads(output.decode('utf-8'))
+        except json.decoder.JSONDecodeError:
+            current_app.logger.warning('Failed to parse json output from rclone lsjson: %s' % output.decode('utf-8'))
+
+        if retcode != 0:
+            current_app.logger.warning(output)
+            current_app.logger.warning(err)
+            raise RuntimeError("Rclone cmd was terminated by signal " + str(retcode) + ": can't run rclone lsjon (stderr: " + str(err) + ")")
+
+        ls_log = str(json_output)
+        ls_log = (ls_log[:15000] + '...') if len(ls_log) > 15000 else ls_log
+        current_app.logger.info('Raw output from rclone lsjson: %s' % ls_log)
+
+        remote_list = []
+        for entry in json_output:
+            if not entry['IsDir']:
+                file_path = entry['Path']
+
+                if file_path.endswith('.rclonelink'):
+                    file_path = file_path[:-11]
+
+                full_file_path = os.path.join(path, file_path)
+
+                if os.path.exists(full_file_path):
+                    remote_list.append({'Path': file_path, 'missing': False})
+                else:
+                    remote_list.append({'Path': file_path, 'missing': True})
```

Comment:
Look at zat, 3 lines economised, make the planet great again

```suggestion
                remote_list.append({'Path': file_path, 'missing': not os.path.exists(full_file_path)})
```